### PR TITLE
[doc] Mention data consistency for categorical features.

### PR DESCRIPTION
**Pull request by trivialfis**
Close https://github.com/dmlc/xgboost/issues/9676 .

### CODE REVIEW

File: demo/guide-python/cat_in_the_dat.py [hunk 1]
```diff
@@ -11,10 +11,12 @@
 And the data can be found at:
 https://www.kaggle.com/shahules/an-overview-of-encoding-techniques/data
 
-Also, see the tutorial for using XGBoost with categorical data:
-:doc:`/tutorials/categorical`.
+  .. versionadded 1.6.0
```
Comment (hcho3) [inline demo/guide-python/cat_in_the_dat.py:14]:
```suggestion
  .. versionadded:: 1.6.0
```

File: doc/tutorials/categorical.rst [hunk 2]
```diff
@@ -131,7 +130,31 @@ types by using the ``feature_types`` parameter in :class:`DMatrix <xgboost.DMatr
...
 feature it's specified as ``"c"``.  The Dask module in XGBoost has the same interface so
-:class:`dask.Array <dask.Array>` can also be used for categorical data.
+:class:`dask.Array <dask.Array>` can also be used for categorical data. Lastly, the
+sklearn interface :py:class:`~xgboost.XGBRegressor` has the same parameter.
+
+****************
+Data Consistency
+****************
+
+XGBoost accepts parameters for which feature is considered categorical, either through the ``dtypes`` of a dataframe or through the ``feature_types`` parameter. However, XGBoost by itself doesn't store information on how categories are encoded in the first place. For instance, given an encoding schema that maps music genre to integer code:
```
Comment (hcho3) [inline doc/tutorials/categorical.rst:140]:
```suggestion
XGBoost accepts parameters to indicate which feature is considered categorical, either through the ``dtypes`` of a dataframe or through the ``feature_types`` parameter. However, XGBoost by itself doesn't store information on how categories are encoded in the first place. For instance, given an encoding schema that maps music genres to integer codes:
```

**Review by hcho3 — CHANGES_REQUESTED** [review]

**Comment by trivialfis** [comment]
Added an example script based on https://github.com/dmlc/xgboost/issues/9676#issuecomment-1768045040 , along with tests.

**Comment by trivialfis** [comment]
@hcho3 Could you please take another look?

**Review by hcho3 — APPROVED** [review]

**Pull request merged**